File: cpp/pengusahaRentalKendaraan.cpp
#pragma once
#include <bits/stdc++.h>
#include <iostream>
#include <string>
#include "orang.cpp"
#include "kendaraan.cpp"

using namespace std;

class pengusahaRentalKendaraan : public orang
{
    private:
        string nama_rental;
        vector <kendaraan*> listKendaraan;

    // CONSTRUCTOR
    public:
        pengusahaRentalKendaraan(){
        }

        pengusahaRentalKendaraan(string noKTP, string nama, string alamat, string nama_rental, vector <kendaraan*> listKendaraan)
        : orang(noKTP, nama, alamat){
            this->nama_rental = nama_rental;
            this->listKendaraan = listKendaraan;
        }

        // DESTRUCTOR
        ~pengusahaRentalKendaraan(){
        }

    // GETTER
    string get_nama_rental(){
        return nama_rental; 
    }

    vector<kendaraan*> get_list_kendaraan(){
        return listKendaraan; 
    }


    // SETTER
    void set_nama_rental(string nama_rental){
        this->nama_rental = nama_rental;
    }

    void set_list_kendaraan(vector <kendaraan*> listKendaraan){
        this->listKendaraan = listKendaraan;
    }

};

File: python/pengusahaRentalKendaraan.py
from orang import orang
from kendaraan import kendaraan

class pengusahaRentalKendaraan(orang):
    def __init__(self, no_ktp="", nama="", alamat="", nama_rental="", list_kendaraan=None):
        super().__init__(no_ktp, nama, alamat)
        self.nama_rental = nama_rental
        self.list_kendaraan = list_kendaraan if list_kendaraan is not None else []

    # Getter methods
    def get_nama_rental(self):
        return self.nama_rental

    def get_list_kendaraan(self):
        return self.list_kendaraan

    # Setter methods
    def set_nama_rental(self, nama_rental):
        self.nama_rental = nama_rental

    def set_list_kendaraan(self, list_kendaraan):
        self.list_kendaraan = list_kendaraan
